module-2/backend/week_two.md

```diff
@@ -5,16 +5,30 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+ - ActiveRecord is Ruby ORM that allows us to interact with/manipulate data in a database as we would a regular Ruby object.
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+ - `belongs_to` and `has_many` are class variables that create association between other models(classes) in ActiveRecord. EX: A Book(model) belongs_to an Author(model), and a Author has_many Books.
 3. What do they allow you to do?
+ - They allow us to create direct connections between different ActiveRecord models.
 4. What's the difference between agile workflow and waterfall method?
+ - Agile is based upon iterative build/test/release cycles; waterfall is based on completing each major phase of a project before moving on to the next (non-iterative).
 5. What is the difference between `#new` and `#create`?
+ - `#new` creates a new object, but does not save it to the database. `#create` creates a new object AND saves it to the database.
 6. At a basic level, what does cURL allow you to do?
+ - cURL allows us to get and send files over different internet file transfer protocols using the command line.
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+ - Not sure if this is supposed to be a trick question... A database with only teachers and students is missing the element that actually creates the relationship between teachers and students, i.e. classes. Simply having a teacher model and a student table would be fairly useless in the normal way that we think about the student/teacher relationship.
```

Comment:
Interetsing! I would say that the relationship still exists at the DB level, though you're right we wouldn't be able to use it in our Rails app without models/has_many/belongs_to type methods. The relationship is created with the foreign key at the **database** level. Our `has_many`/`belongs_to` methods *rely* on that relationship.